docs/examples/ex_extract.rst

```diff
@@ -0,0 +1,456 @@
+'flying' a satellite through a model. The satellite locations are used
+to extract relevant model data enabling direct comparison of observed and
+modeled values.
+
+
+Regular Grid Models
+-------------------
+
+:py:func:`pysatModels.utils.extract.extract_modeled_observations` supports
+extracting values from models on a regular grid. The function can linearly
+inteprolate model values onto instrument locations or use the nearest modeled
+location. Uses :py:func:`scipy.interpolate.interpn` as the underlying
+interpolation function.
+
+Leaving description of @aburrell's function to @aburrell.
+
+
+:py:func:`pysatModels.utils.extract.instrument_view_through_model` supports
+interpolating values from regular grid models onto Instrument locations using
+:py:func:`scipy.interpolate.RegularGridInterpolator`. Consider the following
+example that interpolates model data onto a satellite data set using
+pysat testing data sets.
+
+.. code:: python
+
+   import pysat
+   import pysatModels
+
+   # Load simulated satellite Instrument data set
+   inst = pysat.Instrument('pysat', 'testing', max_latitude=45.)
+   inst.load(2009, 1)
+
+   # Load simulated regular-grid model Instrument
+   model = pysat.Instrument('pysat', 'testmodel')
+   model.load(2009, 1)
+
+Looking at the loaded `model.data` we can see that the model is indeed regular.
+
+.. code:: python
+
+   In []: model.data
+   Out[]:
+   <xarray.Dataset>
+   Dimensions:    (time: 96, latitude: 21, longitude: 73, altitude: 41)
+   Coordinates:
+     * time       (time) datetime64[ns] 2009-01-01 ... 2009-01-01T23:45:00
+     * latitude   (latitude) float64 -50.0 -45.0 -40.0 -35.0 ... 40.0 45.0 50.0
+     * longitude  (longitude) float64 0.0 5.0 10.0 15.0 ... 345.0 350.0 355.0 360.0
+     * altitude   (altitude) float64 300.0 305.0 310.0 315.0 ... 490.0 495.0 500.0
+   Data variables:
+       uts        (time) float64 0.0 900.0 1.8e+03 ... 8.37e+04 8.46e+04 8.55e+04
+       slt        (time, longitude) float64 0.0 0.3333 0.6667 ... 23.08 23.42 23.75
+       mlt        (time, longitude) float64 0.2 0.5333 0.8667 ... 23.28 23.62 23.95
+       dummy1     (time, latitude, longitude) float64 0.0 0.0 0.0 ... 0.0 3.0 6.0
+       dummy2     (time, latitude, longitude, altitude) float64 0.0 0.0 ... 18.0
+
+The `Coordinates` `time`, `latitude`, `longitude`, and `altitude` are all
+one-dimensional and directly relevant to a physical satellite location. The
+equivalent satellite variables are `latitude`, `longitude`, and `altitude`,
+with `time` taken from the Instrument's associated datetimeindex
+(`inst.data.index`).
+
+.. code:: python
+
+   In []: inst.variables
+   Out[]:
+   Index(['uts', 'mlt', 'slt', 'longitude', 'latitude', 'altitude', 'orbit_num',
+          'dummy1', 'dummy2', 'dummy3', 'dummy4', 'string_dummy', 'unicode_dummy',
+          'int8_dummy', 'int16_dummy', 'int32_dummy', 'int64_dummy',
+          'model_dummy2'],
+         dtype='object')
+
+   In []: inst.data.index
+   Out[]:
```

Comment:
We haven't been using this style anywhere else in the docs.  Recommend being consistent.  Either include the output in it's own codeblock or include it as a comment within this code block.